File: services/api/schemas/news.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class NewsItem(BaseModel):
    id: str
    published_at: datetime
    title: str
    summary: str
    source: str
    url: str | None = None


class NewsResponse(BaseModel):
    items: list[NewsItem]
    count: int
    cached: bool

File: frontend/web/src/services/news.ts
import {
  apiGet,
} from "../api/client";


export interface NewsItem {
  id: string;

  published_at: string;

  title: string;

  summary: string;

  source?: string;

  url?: string;
}


export interface NewsResponse {
  items: NewsItem[];

  count: number;

  cached?: boolean;
}


export function getNews(
  signal?: AbortSignal,
): Promise<NewsResponse> {
  return apiGet<NewsResponse>(
    "/api/v1/news?limit=20",
    signal,
  );
}